### DISCUSSION TITLE
Would it be possible to change the sample count?

### DISCUSSION THREAD

**Discussion by FuzzyJJ** [fhdm-dev/scoppy#105] [Q&A] [answered]
I am very pleased with Scoppy for my hobby projects.

Sometimes however I need to test at a fixed sample rate and I found that at 2 or 5 kS/s, the sample count seems to be fixed at 2048 data points which effectively is about 1 or 0.25 s of data. That's too short.  When running at lower sampling rates I can easily store >10k data points so I guess this has something to do with how fast Scoppy can write to memory.

Being able to set a sample count manually would be great. I assume it's also very device dependent what the maximum is and the 2048 is a "safe" number?

**Comment by fhdm-dev** [comment, marked as answer]
Hi,

You are correct. 2048 was chosen because it is a safe number.
Adding the ability to select the sample count has been on my to-do list for a while. Hopefully it won't be too long before it gets implemented :smile: 

Can you do single shot captures instead? They will give you more points but at the sample rates you are using it can take quite a while to do a single capture (manually setting the sample rate might help here - tap the top-left of the screen to do that).

**Reply by FuzzyJJ** [reply]
Thanks! 
The single shot gives me more points and i did indeed fix the sample rate and that way managed to get some usable  results. I'll experiment a bit further.

**Comment by Fabien-B** [comment]
Hello,
I guess is has been implemented since the first post, and I see that the maximum number is 32k for run mode, and 100K for single shot.
At i.e. 500KS/s, that's only 64ms of data! (I am mostly interested by the logic analyzer mode, 500KS/s being quite a low speed for me)

Can you tell the reason behind this number ?

Is this a limitation of the pico? The USB link? An arbitrary limit?

**Reply by fhdm-dev** [reply]
Hi,
The 32k number is fairly arbitrary but it's there because the waveform update rate becomes very low when the number of samples is higher than this. Having said that I guess I could make it higher and let the user decide for themselves if the performance is acceptable.

The 100K limit is because of the available RAM on the pico (RP2040).
